src/main/java/moe/nightfall/vic/integratedcircuits/cp/part/PartWire.java

```diff
@@ -35,10 +35,17 @@ public boolean getOutputToSide(Vec2 pos, ICircuit parent, ForgeDirection side) {
 
 	@Override
 	@SideOnly(Side.CLIENT)
+	// This renders BOTH WIRES AND TUNNELS
 	public void renderPart(Vec2 pos, ICircuit parent, double x, double y, CircuitPartRenderer.EnumRenderType type) {
 		int color = this.getColor(pos, parent);
+		boolean tun = this instanceof PartTunnel;
```

Comment:
There have been a lot of requests to enable wire-like dragging for other parts as well so I think this should be something that's not bound to PartWire, specifically.

Otherwise I pretty much agree with you, and yes, the parts should handle rendering inside the GUI as well, instead of that one giant method I use to do that, mostly replicating already existing render code. But getting started on that would take a lot of time, and I still have a port to 1.8.9 somewhere in the back of my head so I don't want to put too much refactoring effort into this. Its a lot simpler to point these things out in PRs, doesn't mean I'm any better at it myself ;)